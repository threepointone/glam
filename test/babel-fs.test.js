/* eslint-disable no-useless-escape */
/* eslint-env jest */
const babel = require('babel-core')
const fs = require('fs')
const touch = require('touch')
const glamPlugin = require('../src/babel')
const { flush } = require('../src')

jest.mock('fs').mock('touch')

const basic = `
  css\`
    margin: 12px 48px;
    color: #ffffff;
    height: \$\{props => props.height * props.scale\};
    display: flex;
    flex: 1 0 auto;
    color: blue;
    name: class;
    width: \$\{widthVar\};
\``

let output

const cssFilename = __filename + '.css' // eslint-disable-line no-path-concat

describe('babel plugin fs', () => {
  beforeEach(flush)
  test('creates and writes to the css file when it does not exist', () => {
    fs.existsSync.mockReturnValueOnce(false)
    const { code } = babel.transform(basic, {
      plugins: [[glamPlugin, { sync: false }]],
      filename: __filename,
      babelrc: false
    })
    expect(fs.existsSync).toBeCalledWith(cssFilename)
    expect(touch.sync).toBeCalledWith(cssFilename)
    expect(fs.writeFileSync).toHaveBeenCalled()
    expect(fs.writeFileSync.mock.calls[0][0]).toBe(cssFilename)
    expect(fs.writeFileSync.mock.calls[0][1]).toMatchSnapshot()
    output = fs.writeFileSync.mock.calls[0][1]
    expect(code).toMatchSnapshot()
  })
  test('writes to the css file when it does exist ', () => {
    fs.existsSync.mockReturnValueOnce(true)
    fs.readFileSync.mockReturnValueOnce('')
    const { code } = babel.transform(basic, {
      plugins: [[glamPlugin, { sync: false }]],
      filename: __filename,
      babelrc: false
    })
    expect(fs.existsSync).toBeCalledWith(cssFilename)
    expect(touch.sync).toHaveBeenCalledTimes(1)
    expect(fs.writeFileSync).toHaveBeenCalledTimes(2)
    expect(fs.writeFileSync.mock.calls[1][0]).toBe(cssFilename)
    expect(fs.writeFileSync.mock.calls[1][1]).toMatchSnapshot()
    expect(code).toMatchSnapshot()
  })
  test('does not write to the css file when it is the same as is already written', () => {
    fs.existsSync.mockReturnValueOnce(true)
    fs.readFileSync.mockReturnValueOnce(output)
    const { code } = babel.transform(basic, {
      plugins: [[glamPlugin, { sync: false }]],
      filename: __filename,
      babelrc: false
    })
    expect(fs.existsSync).toBeCalledWith(cssFilename)
    expect(touch.sync).toHaveBeenCalledTimes(1)
    expect(fs.writeFileSync).toHaveBeenCalledTimes(2)
    expect(code).toMatchSnapshot()
  })
})
