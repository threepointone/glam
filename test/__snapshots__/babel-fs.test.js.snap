// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin fs creates and writes to the css file when it does not exist 1`] = `
"/* do not edit this file */
.class-wf93qd { margin: 12px 48px;
    color: #ffffff;
    height: var(--class-wf93qd-0);
    display: -webkit-box;
    display: -moz-box;
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
    color: blue;
    name: class;
    width: var(--class-wf93qd-1); }"
`;

exports[`babel plugin fs creates and writes to the css file when it does not exist 2`] = `
"
css(\\"class-wf93qd\\", [props => props.height * props.scale, widthVar]);"
`;

exports[`babel plugin fs does not write to the css file when it is the same as is already written 1`] = `
"
css(\\"class-wf93qd\\", [props => props.height * props.scale, widthVar]);"
`;

exports[`babel plugin fs writes to the css file when it does exist  1`] = `
"/* do not edit this file */
.class-wf93qd { margin: 12px 48px;
    color: #ffffff;
    height: var(--class-wf93qd-0);
    display: -webkit-box;
    display: -moz-box;
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
    color: blue;
    name: class;
    width: var(--class-wf93qd-1); }"
`;

exports[`babel plugin fs writes to the css file when it does exist  2`] = `
"
css(\\"class-wf93qd\\", [props => props.height * props.scale, widthVar]);"
`;
